packages/vm/src/wasm_backend/store.rs

```diff
@@ -42,47 +43,72 @@ pub fn make_compile_time_store(
 
     #[cfg(feature = "cranelift")]
     {
-        let mut config = Cranelift::default();
+        let mut compiler = Cranelift::default();
         for middleware in middlewares {
-            config.push_middleware(middleware.clone());
+            compiler.push_middleware(middleware.clone());
         }
-        config.push_middleware(deterministic);
-        config.push_middleware(metering);
-        let engine = Universal::new(config).engine();
-        make_store_with_engine(&engine, memory_limit)
+        compiler.push_middleware(deterministic);
+        compiler.push_middleware(metering);
+        make_store_with_engine(compiler.into(), memory_limit)
     }
 
     #[cfg(not(feature = "cranelift"))]
     {
-        let mut config = Singlepass::default();
+        let mut compiler = Singlepass::default();
         for middleware in middlewares {
-            config.push_middleware(middleware.clone());
+            compiler.push_middleware(middleware.clone());
         }
-        config.push_middleware(deterministic);
-        config.push_middleware(metering);
-        let engine = Universal::new(config).engine();
-        make_store_with_engine(&engine, memory_limit)
+        compiler.push_middleware(deterministic);
+        compiler.push_middleware(metering);
+        make_store_with_engine(compiler.into(), memory_limit)
+    }
+}
+
+/// Creates an engine with the default compiler.
+pub fn make_engine(middlewares: &[Arc<dyn ModuleMiddleware>]) -> Engine {
+    let gas_limit = 0;
+    let deterministic = Arc::new(Gatekeeper::default());
+    let metering = Arc::new(Metering::new(gas_limit, cost));
+
+    #[cfg(feature = "cranelift")]
+    {
+        let mut compiler = Cranelift::default();
+        for middleware in middlewares {
+            compiler.push_middleware(middleware.clone());
+        }
+        compiler.push_middleware(deterministic);
+        compiler.push_middleware(metering);
+        compiler.into()
+    }
+
+    #[cfg(not(feature = "cranelift"))]
+    {
+        let mut compiler = Singlepass::default();
+        for middleware in middlewares {
+            compiler.push_middleware(middleware.clone());
+        }
+        compiler.push_middleware(deterministic);
+        compiler.push_middleware(metering);
+        compiler.into()
     }
```

Comment:
nit: Not really related to this PR, but this could be shortened like this:
```suggestion
    #[cfg(feature = "cranelift")]
    let mut compiler = Cranelift::default();
    #[cfg(not(feature = "cranelift"))]
    let mut compiler = Singlepass::default();

    for middleware in middlewares {
        compiler.push_middleware(middleware.clone());
    }
    compiler.push_middleware(deterministic);
    compiler.push_middleware(metering);
    compiler.into()
```
and same in `make_compile_time_store`